core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1270,6 +1267,50 @@ class SnappySession(@transient private val sc: SparkContext,
     }
   }
 
+  def invalidateAll(): Unit = {
+    sessionCatalog.invalidateAll()
+  }
+
+  private[sql] def alterTable(tableName: String, isAddColumn: Boolean,
+                 column: StructField): Unit = {
+    alterTable(sessionCatalog.newQualifiedTableName(tableName), isAddColumn, column)
+  }
+
+  private[sql]  def alterTable(tableIdent: QualifiedTableName, isAddColumn: Boolean,
+                              column: StructField): Unit = {
+    val plan = try {
+      sessionCatalog.lookupRelation(tableIdent)
+    } catch {
+      case tnfe: TableNotFoundException =>
+        throw tnfe
+    }
+    plan match {
+      case LogicalRelation(c: ColumnFormatRelation, _, _) =>
+        throw new AnalysisException("alter table not supported for column tables")
+      case _ =>
+    }
+
+    SnappyContext.getClusterMode(sc) match {
+      case ThinClientConnectorMode(_, _) =>
+        val isTempTable = sessionCatalog.isTemporaryTable(tableIdent)
+        if (!isTempTable) {
+          sessionCatalog.invalidateTable(tableIdent)
+          sessionCatalog.asInstanceOf[ConnectorCatalog].connectorHelper
+            .alterTable(tableIdent, isAddColumn, column)
+          return
+        }
+      case _ =>
+    }
+
+    plan match {
+      case LogicalRelation(ar: AlterableRelation, _, _) =>
```

Comment:
In fcat if AlterableRelation is there and which I guess only relations of rowtables must be implementing then we should just check for that and then call the same on session which depending on the mode should either fire the procedure or call the alter table on the relation.